model/version_activation.go

```diff
@@ -32,7 +39,83 @@ func DoProjectActivation(ctx context.Context, id string, ts time.Time) (bool, er
 	}
 
 	return activated, nil
+}
+
+func activateEveryRecentMainlineCommitForProject(ctx context.Context, projectRef *ProjectRef, ts time.Time) (bool, error) {
+	lastActivatedVersion, err := VersionFindOne(ctx, VersionByMostRecentActivated(projectRef.Id, ts))
+	if err != nil {
+		return false, errors.Wrap(err, "finding most recently activated version")
+	}
+
+	var activateVersions []Version
+	if lastActivatedVersion == nil {
+		// No previously activated versions - this might be a new project or first activation
+		// Activate ALL unactivated non-ignored versions to ensure complete coverage
+		activateVersions, err = VersionFind(ctx, VersionsAllUnactivatedNonIgnored(projectRef.Id, ts, projectRef.RunEveryMainlineCommitLimit))
+		if err != nil {
+			return false, errors.Wrapf(err, "finding all unactivated non-ignored versions")
+		}
+	} else {
+		// Find all unactivated versions since the last activated one
+		activateVersions, err = VersionFind(ctx, VersionsUnactivatedSinceLastActivated(projectRef.Id, ts, lastActivatedVersion.RevisionOrderNumber, projectRef.RunEveryMainlineCommitLimit))
+		if err != nil {
+			return false, errors.Wrapf(err, "finding unactivated versions since last activated version '%s'", lastActivatedVersion.Id)
+		}
+	}
+
+	if len(activateVersions) == 0 {
+		grip.Info(message.Fields{
+			"message":   "no versions to activate for repository",
+			"project":   projectRef.Id,
+			"operation": "project-activation-every-commit",
+		})
+		return false, nil
+	}
+	// Activate all eligible versions
+	anyActivated := false
+	activatedCount := 0
+	for _, version := range activateVersions {
+		activated, err := ActivateElapsedBuildsAndTasks(ctx, &version)
+		if err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":   "error activating version",
+				"project":   projectRef.Id,
+				"version":   version.Id,
+				"revision":  version.Revision,
+				"operation": "project-activation-every-commit",
+			}))
+			// Continue with other versions even if one fails
+			continue
+		}
+		if activated {
+			anyActivated = true
+			activatedCount++
+			grip.Info(message.Fields{
+				"message":   "activated version",
+				"project":   projectRef.Id,
+				"version":   version.Id,
+				"revision":  version.Revision,
+				"operation": "project-activation-every-commit",
+			})
+		}
+	}
+
+	if anyActivated {
```

Comment:
Similarly I'd rather not have this logic just for logging, unless you intend to get rid of it. Maybe we could keep the info log but drop last_activated_version? I feel like this logic also doesn't quite do what you want, since lastActivatedVersion isn't set to anything in the if case at line 51. What are we trying to achieve with this logging?

Reply:
Also you don't need anyActivated; you can just use activatedCount > 0